api/data_explorer/controllers/facets_controller.py

```diff
@@ -79,27 +79,27 @@ def facets_get(filter=None, extraFacets=None):  # noqa: E501
     :type extraFacets: List[str]
     :rtype: FacetsResponse
     """
-    extra_es_facets, extra_ui_facets = _process_extra_facets(extraFacets)
-    combined_es_facets = OrderedDict(extra_es_facets.items() + current_app.
-                                     config['ELASTICSEARCH_FACETS'].items())
+    extra_ui_facets = _process_extra_facets(extraFacets)
     combined_ui_facets = OrderedDict(extra_ui_facets.items() +
                                      current_app.config['UI_FACETS'].items())
     search = DatasetFacetedSearch(
-        elasticsearch_util.get_facet_value_dict(filter, combined_es_facets),
-        combined_es_facets)
+        elasticsearch_util.get_facet_value_dict(filter, combined_ui_facets),
```

Comment:
I would say we should change `filter` since its a built-in function, but this is dictated by the API so I guess it's fine to leave. 

@melissachang?

Reply:
I would leave

Reply:
(The [Python style guide](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#3161-names-to-avoid) doesn't say anything about avoiding names that are built-in functions. In general, across languages, I feel like the rule of thumb regarding reserved words is: if it compiles, it's ok.)

Reply:
I'd disagree, just because something compiles doesn't mean it's the right thing to do. Technically you can define a variable `list = 'foo'` in python, but that means you are overshadowing the built-in `list()` function and it can't be used elsewhere in that context. `filter` is exactly the same, it prevents us from using `filter()` anywhere in this context. Plus it highlights incorrectly in IDEs/Github.

Even though the style guide doesn't explicitly state this, I think where possible we should avoid using built-in function names. In this case it would require changing our API so I'm definitely fine leaving it.